dtfs-central-api/src/services/utilisation-report-data-validator/index.ts

```diff
@@ -99,38 +99,36 @@ export const validateUtilisationReportCsvCellData = (
     generatePaymentExchangeRateError,
   ];
 
-  return csvData.flatMap((csvRow) => {
-    const csvDataErrors: UtilisationReportDataValidationError[] = [];
-
-    cellValidations.forEach(({ header, errorGenerator }) => {
-      if (availableHeaders.includes(header)) {
-        const error = errorGenerator(csvRow[header], csvRow.exporter?.value);
-        if (error) {
-          csvDataErrors.push(error);
+  const allErrors = await Promise.all(
+    csvData.map(async (csvRow) => {
+      const cellErrorPromises = cellValidations.map(({ header, errorGenerator }) => {
+        if (!availableHeaders.includes(header)) {
+          return null;
         }
-      }
-    });
 
-    rowValidations.forEach((errorGenerator) => {
-      const error = errorGenerator(csvRow);
-      if (error) {
-        csvDataErrors.push(error);
-      }
-    });
+        return errorGenerator(csvRow[header], csvRow.exporter?.value);
+      });
 
-    return csvDataErrors;
-  });
+      const rowErrorPromises = rowValidations.map((errorGenerator) => errorGenerator(csvRow));
+
+      return Promise.all([...cellErrorPromises, ...rowErrorPromises]);
+    }),
+  );
+
+  const allNonNullErrors = allErrors.flat().filter((error): error is UtilisationReportDataValidationError => error !== null);
```

Comment:
Maybe could be just be thefollowing?:

```suggestion
  const allNonNullErrors = allErrors.flat().filter((error) => error !== null);
```

Also i'm wondering if something like `cleanErrors` or `filteredErrors` would be easier to read than `allNonNullErrors`

Reply:
Inlined so we don't have to rename it